assets/src/blocks/SocialMedia/SocialMediaEditorScript.js

```diff
@@ -0,0 +1,245 @@
+import { useEffect } from '@wordpress/element';
+import { InspectorControls } from '@wordpress/block-editor';
+import {
+  RadioControl,
+  SelectControl,
+  PanelBody,
+} from '@wordpress/components';
+import { SocialMediaEmbed } from './SocialMediaEmbed';
+import { URLInput } from '../../components/URLInput/URLInput';
+import { HTMLSidebarHelp } from '../../components/HTMLSidebarHelp/HTMLSidebarHelp';
+import {
+  OEMBED_EMBED_TYPE,
+  FACEBOOK_EMBED_TYPE,
+  FACEBOOK_PAGE_TAB_TIMELINE,
+  FACEBOOK_PAGE_TAB_EVENTS,
+  FACEBOOK_PAGE_TAB_MESSAGES,
+  ALLOWED_OEMBED_PROVIDERS,
+} from './SocialMediaConstants.js';
+
+const { RichText } = wp.blockEditor;
+const { __ } = wp.i18n;
+const { apiFetch } = wp;
+const { addQueryArgs } = wp.url;
+
+const loadScriptAsync = uri => {
```

Comment:
We could also write this as an `async` function, and use `await`. `useEffect` cannot receive an `async` function, however you can invoke one inside of it. For example: https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/902e5e4322269414a2def75e6399cd679168b61a/assets/src/components/Sidebar/NewThemeSettings.js#L15-L20

Reply:
Updated!

Reply:
:+1: I was first also thinking about this function in my comment, however on closer look it might be tricky to write an `async` equivalent of what `tag.onload= () => {resolve()}` does :thinking: